app/src/main/java/ar/com/wolox/android/cookbook/instagramlogin/InstagramLoginRecipeFragment.kt

```diff
@@ -0,0 +1,118 @@
+package ar.com.wolox.android.cookbook.instagramlogin
+
+import android.view.View
+import android.webkit.CookieManager
+import android.widget.Toast
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import ar.com.wolox.android.cookbook.R
+import ar.com.wolox.android.cookbook.instagramlogin.adapter.InstagramLoginAuthListener
+import ar.com.wolox.android.cookbook.instagramlogin.adapter.InstagramLoginView
+import ar.com.wolox.android.cookbook.instagramlogin.model.InstagramDataItem
+import ar.com.wolox.wolmo.core.fragment.WolmoFragment
+import kotlinx.android.synthetic.main.fragment_instagram_login.*
+
+class InstagramLoginRecipeFragment : WolmoFragment<InstagramLoginRecipePresenter>(), InstagramLoginRecipeView {
+
+    private lateinit var viewAdapter: InstagramLoginRecipeAdapter
+    private var viewManager: RecyclerView.LayoutManager = LinearLayoutManager(context)
+    private lateinit var pictureList: MutableList<InstagramDataItem>
+
+    override fun layout(): Int = R.layout.fragment_instagram_login
+
+    override fun init() {
+    }
+
+    override fun setListeners() {
+        super.setListeners()
+
+        vSessionBtn.setOnClickListener {
+            presenter.onSessionButtonClicked()
+        }
+
+        vDataBtn.setOnClickListener {
+            presenter.onFetchDataButtonClicked()
+        }
+    }
+
+    override fun enableLoginBtn() {
+        vSessionBtn.text = getString(R.string.instagram_login_btn)
+    }
+
+    override fun enableLogoutBtn() {
+        vSessionBtn.text = getString(R.string.instagram_logout_btn)
+    }
+
+    override fun igLogout() {
+        CookieManager.getInstance().removeAllCookies { p0 ->
+            presenter.onLogoutResponse(p0)
+        }
+    }
+
+    override fun showLoginError() {
+        Toast.makeText(context, getString(R.string.instagram_error_login), Toast.LENGTH_LONG).show()
```

Comment:
We can make something like showError( ErrorType ) and then the view decides what message to show